packages/tokens/scripts/extract-carbon-parts/extractThemes.ts

```diff
@@ -0,0 +1,135 @@
+
+  // notification tokens
+  const carbonThemesNotificationTokensDtcg = convertObjectToDtcgFormat({ value: cleanupObj(notificationTokens), type: 'color', group: 'cds-themes' });
+  await saveCarbonDtcgTokensAsJsonFile({ obj: carbonThemesNotificationTokensDtcg, group: 'themes', file: 'notification-tokens' });
+
+  // status tokens
+  const carbonThemesStatusTokensDtcg = convertObjectToDtcgFormat({ value: cleanupObj(statusTokens), type: 'color', group: 'cds-themes' });
+  await saveCarbonDtcgTokensAsJsonFile({ obj: carbonThemesStatusTokensDtcg, group: 'themes', file: 'status-tokens' });
+
+  // status tokens
+  const carbonThemesTagTokensDtcg = convertObjectToDtcgFormat({ value: cleanupObj(tagTokens), type: 'color', group: 'cds-themes' });
+  await saveCarbonDtcgTokensAsJsonFile({ obj: carbonThemesTagTokensDtcg, group: 'themes', file: 'tag-tokens' });
+
+  // themes (this is a special case, )
+  const carbonThemesAllTokensDtcg = convertThemeObjectToDtcg(cleanupObj(themes), '[root]');
+  await saveCarbonDtcgTokensAsJsonFile({ obj: carbonThemesAllTokensDtcg, group: 'themes', file: 'themes' });
+
+}
+
+// function that recursively iterates on an object and
+// - replaces any key named 'whiteTheme' with 'white'
+// - removes entries whose value is an array (`breakpoints`)
+
+function cleanupObj(obj: Record<string, any>): Record<string, any> {
+  // Base case: if obj is not an object or is null, return it as is
+  if (typeof obj !== 'object' || obj === null) {
+    return obj;
+  }
+
+  // For arrays, map over each element and process recursively
+  if (Array.isArray(obj)) {
+    return obj.map(item => cleanupObj(item));
+  }
+
+  // For objects, create a new object with potentially renamed keys
+  const result: Record<string, any> = {};
+
+  for (const [key, value] of Object.entries(obj)) {
+    // Skip entries whose value is an array
+    if (Array.isArray(value)) {
+      continue;
+    }
+
+    // Rename 'whiteTheme' key to 'white'
+    const newKey = key === 'whiteTheme' ? 'white' : key;
+
+    // Recursively process nested objects
+    result[newKey] = cleanupObj(value);
+  }
+
+  return result;
+}
+
+function convertThemeObjectToDtcg(obj: Record<string, any>, key: string) {
+  if (typeof obj === 'object') {
+    // it's an object, process all its keys
+    for (const key in obj) {
+      if (Object.prototype.hasOwnProperty.call(obj, key)) {
+        obj[key] = convertThemeObjectToDtcg(obj[key], key);
+      }
+    }
+    return obj;
+  } else {
+    // it's a leaf, convert it
+    return convertThemeValueToDtcg(obj, key);
+  }
+}
+
+function convertThemeValueToDtcg(value: string | number, key: string) {
+  let type;
+  // first try inferring the type from known keys
+  switch (key) {
+    case 'fontFamily':
+      type = 'font-family';
+      break;
+    case 'fontSize':
+      type = 'font-size';
+      break;
+    case 'fontWeight':
+      type = 'font-weight';
+      break;
+    case 'lineHeight':
+      type = 'lineHeight';
```

Comment:
```suggestion
      type = 'line-height';
```
[Suggestion] In the `themes.json` I notice the `lineHeight` type is getting set to `unknown`. I'm not sure if it's due to this line using camel case instead of kebab, but it could be.

From `themes.json`: 
```
"lineHeight": {
  "$type": "unknown",
  "$value": 1.42857,
  "group": "cds-themes",
  "private": true,
  "cds-original-value": 1.42857
},
```

Reply:
fixed, see https://github.com/hashicorp/design-system/pull/3169/commits/7ecb2d300f961fd8bc875f9651da5d51bf1e846d